tools/distpartitioning/multi_dev_init.py

```diff
@@ -0,0 +1,499 @@
+            if ((output_list[idx] is not None) and (ntype_name+'/feat' in output_list[idx])):
+                if (ntype_name+'/feat' not in rcvd_node_features):
+                    rcvd_node_features[ntype_name+'/feat'] = torch.empty((0,), dtype=torch.float)
+                    rcvd_global_nids[ntype_name+'/feat'] = torch.empty((0,), dtype=torch.int64)
+                rcvd_node_features[ntype_name+'/feat'] = \
+                    torch.cat((rcvd_node_features[ntype_name+'/feat'], output_list[idx][ntype_name+'/feat']))
+                rcvd_global_nids[ntype_name+'/feat'] = \
+                    np.concatenate((rcvd_global_nids[ntype_name+'/feat'], output_nid_list[idx][ntype_name+'/feat']))
+    end = timer()
+    print('[Rank: ', rank, '] Total time for node feature exchange: ', timedelta(seconds = end - start))
+
+    return rcvd_node_features, rcvd_global_nids
+
+def exchange_graph_data(rank, world_size, node_data, node_features, edge_data,
+        node_part_ids, ntypes_map, ntypes_nid_map, ntype_id_count):
+    """
+    Wrapper function which is used to shuffle graph data on all the processes. 
+
+    Parameters: 
+    -----------
+    rank : int
+        rank of the current process
+    world_size : int
+        total no. of participating processes. 
+    node_data : dictionary
+        dictionary where node data is stored, which is initially read from the nodes txt file mapped
+        to the current process
+    node_feautres: dicitonary
+        dictionry where node_features are stored and this information is read from the appropriate
+        node features file which belongs to the current process
+    edge_data : dictionary
+        dictionary which is used to store edge information as read from the edges.txt file assigned
+        to each process.
+    node_part_ids : numpy array
+        numpy array which store the partition-ids and indexed by global_nids
+    ntypes_map : dictionary
+        mappings between node type names and node type ids
+    ntypes_nid_map : dictionary
+        mapping between node type names and global_nids which belong to the keys in this dictionary
+    ntype_id_count : dictionary
+        mapping between node type id and no of nodes which belong to each node_type_id
+    """
+    rcvd_node_features, rcvd_global_nids = exchange_node_features(rank, world_size, node_data, \
+            node_features, ntypes_map, ntypes_nid_map, ntype_id_count, node_part_ids)
+    print( 'Rank: ', rank, ' Done with node features exchange.')
+
+    exchange_node_data(rank, world_size, node_data)
+    exchange_edge_data(rank, world_size, edge_data)
+    return rcvd_node_features, rcvd_global_nids
+
+def read_dataset(rank, world_size, node_part_ids, params):
+    """
+    After reading input graph files, add additional information(columns) are added
+    to these data structures, as discussed in detail in the initialize.py file for the
+    case of single-file-format dataset use-case. 
```

Comment:
this function is used for single file format?

Reply:
For single file format, there is a separate function, which was present in the initialize.py. But per our discussion, we decided that from now on we only support multiple file format. So that code was deleted.